### ISSUE TITLE
FR: SSH recording retention settings for large transfers

### ISSUE THREAD

**Issue by northys** [tailscale/tailscale#8177]
### What is the issue?

Got out of space 3 times in a row and always fixed by removing all SSH sessions. So I did a bit of investigation today and foud out that the SSH session which has 20 gigabites has this header.

```
{"version":2,"width":0,"height":0,"timestamp":1684576211,"env":{"TERM":"xterm-256color"},"command":"sudo rsync --server --sender -logDtpAXre.iLsfxCIvu . /","srcNode":"backup.tailMASKED.ts.net","srcNodeID":"nhxu7n1CNTRL","srcNodeTags":["tag:backup","tag:server"],"sshUser":"backup","localUser":"backup"}
```

Is there any reason why rsync session should be recorded AND take so much space? Beucase it is rsync triggered by our backup server... but the backup is running this way for past half year and first time I noticed out of disk space because of tailscale session recording filled up disk space was this thursday...

### Are there any recent changes that introduced the issue?

```
root@masked /var/log # cat dpkg.log | grep tailscale
2023-05-09 18:35:12 upgrade tailscale:amd64 1.36.2 1.40.0
2023-05-09 18:35:12 status half-configured tailscale:amd64 1.36.2
2023-05-09 18:35:12 status unpacked tailscale:amd64 1.36.2
2023-05-09 18:35:12 status half-installed tailscale:amd64 1.36.2
2023-05-09 18:35:13 status half-installed tailscale:amd64 1.36.2
2023-05-09 18:35:14 status unpacked tailscale:amd64 1.40.0
2023-05-09 18:35:14 status unpacked tailscale:amd64 1.40.0
2023-05-09 18:35:14 install tailscale-archive-keyring:all <none> 1.35.181
2023-05-09 18:35:14 status half-installed tailscale-archive-keyring:all 1.35.181
2023-05-09 18:35:14 status unpacked tailscale-archive-keyring:all 1.35.181
2023-05-09 18:35:14 status unpacked tailscale-archive-keyring:all 1.35.181
2023-05-09 18:35:14 configure tailscale:amd64 1.40.0 <none>
2023-05-09 18:35:14 status unpacked tailscale:amd64 1.40.0
2023-05-09 18:35:14 status unpacked tailscale:amd64 1.40.0
2023-05-09 18:35:14 status half-configured tailscale:amd64 1.40.0
2023-05-09 18:35:16 status installed tailscale:amd64 1.40.0
2023-05-09 18:35:16 configure tailscale-archive-keyring:all 1.35.181 <none>
2023-05-09 18:35:16 status unpacked tailscale-archive-keyring:all 1.35.181
2023-05-09 18:35:16 status half-configured tailscale-archive-keyring:all 1.35.181
2023-05-09 18:35:16 status installed tailscale-archive-keyring:all 1.35.181
2023-05-16 09:13:14 upgrade tailscale:amd64 1.40.0 1.40.1
2023-05-16 09:13:14 status half-configured tailscale:amd64 1.40.0
2023-05-16 09:13:14 status unpacked tailscale:amd64 1.40.0
2023-05-16 09:13:14 status half-installed tailscale:amd64 1.40.0
2023-05-16 09:13:14 status half-installed tailscale:amd64 1.40.0
2023-05-16 09:13:16 status unpacked tailscale:amd64 1.40.1
2023-05-16 09:13:16 status unpacked tailscale:amd64 1.40.1
2023-05-16 09:13:16 configure tailscale:amd64 1.40.1 <none>
2023-05-16 09:13:16 status unpacked tailscale:amd64 1.40.1
2023-05-16 09:13:16 status unpacked tailscale:amd64 1.40.1
2023-05-16 09:13:16 status half-configured tailscale:amd64 1.40.1
2023-05-16 09:13:18 status installed tailscale:amd64 1.40.1
```

### OS version

Ubuntu 16.04

### Tailscale version

1.40.1

### Bug report

BUG-12e9b18341f2872803c0c69a1b48d75844881f54bf279d2a27c65f2b89601a8f-20230520180232Z-fe7ab319dcaaa239

**Comment by DentonGentry** [comment]
> Is there any reason why rsync session should be recorded AND take so much space?

It currently records everything. We can look at whether it should have limits like only the first portion of an rsync, or perhaps discard such large records after N days.

**Comment by northys** [comment]
@DentonGentry there must have been a change introduced because as I said the server is backed up by rsync multiple times a day for a quite a while... and tailscale is there since January... and since the disk has only 32 gigabites I would notice the leak much faster because the tailscale actually consumed 40% of disk in few minutes...

**Comment by DentonGentry** [comment]
Session recording was introduced May 11. https://tailscale.com/changelog/#2023-05-11-service

**Comment by northys** [comment]
```
2023-05-16 09:13:14 upgrade tailscale:amd64 1.40.0 1.40.1
```

The sudden peaks in disk usage started the day after upgrade to 10.40.1...
![image](https://github.com/tailscale/tailscale/assets/2878126/0e98374e-639f-420f-a928-a0d30ce65243)

**Comment by northys** [comment]
@DentonGentry we were using the "alpha recorder" for months by enabling the mendioned flag

**Comment by DentonGentry** [comment]
Local log storage was supported until 1.38.3.
It stopped working in 1.38.4 [while working on log streaming](https://github.com/tailscale/tailscale/pull/7640).
Local log storage was fixed again in 1.40.1: https://tailscale.com/changelog/#2023-05-10-client

**Comment by maisem** [comment]
This might be because we started logging non-interactive sessions.
